libraries/AP_OSD/AP_OSD.cpp

```diff
@@ -190,23 +194,29 @@ AP_OSD::AP_OSD()
         AP_HAL::panic("AP_OSD must be singleton");
     }
     AP_Param::setup_object_defaults(this, var_info);
+#if OSD_ENABLED
     // default first screen enabled
     screen[0].enabled = 1;
+    previous_pwm_screen = -1;
+#endif
 #ifdef WITH_SITL_OSD
-    osd_type.set_default(2);
+    osd_type.set_default(OSD_SITL);
 #endif
 
 #ifdef HAL_OSD_TYPE_DEFAULT
     osd_type.set_default(HAL_OSD_TYPE_DEFAULT);
+#elif OSD_PARAM_ENABLED
+    osd_type.set_default(OSD_TX);   // the OSD only exists on the radio transmitter
```

Comment:
doesn't seem right. I don't think we can default this safely

Reply:
Fixed